components/Auth/SignIn.js

```diff
@@ -61,12 +62,71 @@ const styles = {
 class SignIn extends Component {
   constructor (props) {
     super(props)
+
     this.state = {
       email: props.email || '',
       polling: false,
       loading: false,
       success: undefined
     }
+
+    this.onFormSubmit = (event) => {
+      event.preventDefault()
+
+      const { loading, error } = this.state
+
+      if (error) {
+        this.setState(() => ({ dirty: true }))
+        return
+      }
+
+      if (loading) {
+        return
+      }
+
+      this.setState(() => ({ loading: true }))
+
+      runInApp(this.signInApp, this.signInBrowser)
```

Comment:
I think is cool to have a helper function that works as an 'if' statement egarding the user is in App or not. I don't know in this scenario if keeping the `runInApp` name is the best